### PR TITLE
🎉 Source Orb: support enriching ledger entries with numeric properties

### PR DESCRIPTION
**Pull request by kgrover**
## What
This adds a configuration parameter to the Orb Source connector, `numeric_event_properties_keys`, which allows the user to list keys whose property values are numeric (as opposed to string-valued). The dynamic schema generation in the connector constructs the schema accordingly.

🐛 This also makes a few schema updates to the existing declared schema, as well as a minor bug fix to output `id` instead of `event_id` when outputting an enriched event.

## How
When we dynamically generate the JSON schema, enriching `credit_ledger_entries.json`, we treat the keys passed in for `numeric_event_properties_keys` separately.

## Recommended reading order
1. `spec.json` and `source.py` for the new addition to support numeric properties
2. Schema files for schema changes

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

This connector should currently be unused, so do not expect any user impact.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

## Tests

<details><summary><strong>Unit</strong></summary>

orb_source_unit_tests

</details>

<details><summary><strong>Acceptance</strong></summary>
Screen Shot 2022-03-03 at 9 38 03 AM

</details>

### CODE REVIEW

**Review by marcosmarxm — APPROVED** [review]
thanks @kgrover

**Comment by marcosmarxm** [comment]
@misteryeo can you review the change in spec.json of this contribution?

**Comment by marcosmarxm** [comment]
@kgrover can you bump the connector version in `airbyte-config/init/src/main/resources` and the Dockerfile inside the container folder?
See more info here: https://docs.airbyte.com/connector-development#4.-publish-the-connector

**Review by lmossman — APPROVED** [review]

**Comment by lmossman** [comment]
Looking at that ^ action, it seems that the test actually passed but the github action just failed to update the above comment with the success status. So I think we can consider that to be a success

**Comment by lmossman** [comment]
/publish connector=connectors/source-orb repo=orbcorp/airbyte

**Comment by lmossman** [comment]
/publish connector=connectors/source-orb repo=orbcorp/airbyte

**Comment by lmossman** [comment]
/publish connector=connectors/source-orb repo=orbcorp/airbyte

**Review by misteryeo — APPROVED** [review]
LGTM!

File: docs/integrations/sources/orb.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ an Orb Account and API Key.
 ## Changelog
 
 | Version | Date | Pull Request | Subject |
+| 0.2.0 | 2022-03-03 | [10839](https://github.com/airbytehq/airbyte/pull/10839) | Support ledger entries with numeric properties + schema fixes
```
Comment (marcosmarxm) [inline docs/integrations/sources/orb.md:54]:
```suggestion
| 0.1.1 | 2022-03-03 | [10839](https://github.com/airbytehq/airbyte/pull/10839) | Support ledger entries with numeric properties + schema fixes
```

**Review by marcosmarxm — APPROVED** [review]
It's finished @kgrover. Can you just correct the documetation and run `./gradlew airbyte-config:init:processResources`?

**Pull request merged**